sulat/scoring.py

```
@@ -0,0 +1,540 @@
+    # Add default scheme if missing
+    if not s.startswith(('http://', 'https://')):
+        s = 'https://' + s
+    
+    # Convert to lowercase (for domain)
+    parts = s.split('://', 1)
+    if len(parts) == 2:
+        scheme, rest = parts
+        # Split at first slash after scheme to separate domain/path
+        if '/' in rest:
+            domain, path = rest.split('/', 1)
+            domain = domain.lower().lstrip('www.')
```

Ruff (0.13.1)

Using `.strip()` with multi-character strings is misleading

(B005)